openwisp_controller/mixins.py

```diff
@@ -35,3 +38,14 @@ class RelatedDeviceProtectedAPIMixin(
 
 class ProtectedAPIMixin(BaseProtectedAPIMixin, FilterByOrganizationManaged):
     pass
+
+
+class AutoRevisionMixin(RevisionMixin):
+    def dispatch(self, request, *args, **kwargs):
+        if request.method in ('GET', 'HEAD', 'OPTIONS'):
+            return super().dispatch(request, *args, **kwargs)
+        with reversion.create_revision():
```

Comment:
Wouldn't this trigger also on DELETE? May not be desired, since DELETE wouldn't have any change and would store an additional revision identical to the previous one.

I would do the opposite: trigger only on PUT/PATCH/POST. We need a test for each HTTP method.
POST may have to create the initial revision.